File: Java/ComprobantePago.java
import java.util.Date;

class ComprobantePago {
    private static int correlativoID = 1;
    private int idComprobante;
    private TipoComprobante tipoDocumento;
    private String ruc;
    private String razonSocial;
    private String numeroSerial;
    private Date fechaEmision;
    private double montoTotal;
    private double subtotal;
    private double igv;
    private double total;

    private SolicitudGasto solicitud; // get set y copia

    // Constructor
    public ComprobantePago(TipoComprobante tipoDocumento, String ruc,
                           String razonSocial, String numeroSerial, Date fechaEmision,
                           double montoTotal, double subtotal, double igv, double total) {
        this.idComprobante = this.correlativoID++;
        this.tipoDocumento = tipoDocumento;
        this.ruc = ruc;
        this.razonSocial = razonSocial;
        this.numeroSerial = numeroSerial;
        this.fechaEmision = fechaEmision;
        this.montoTotal = montoTotal;
        this.subtotal = subtotal;
        this.igv = igv;
        this.total = total;
    }

    // Selectores
    public int getIdComprobante() {
        return idComprobante;
    }
    public TipoComprobante getTipoDocumento() {
        return tipoDocumento;
    }
    public String getRuc() {
        return ruc;
    }
    public String getRazonSocial() {
        return razonSocial;
    }
    public String getNumeroSerial() {
        return numeroSerial;
    }
    public Date getFechaEmision() {
        return fechaEmision;
    }
    public double getMontoTotal() {
        return montoTotal;
    }
    public double getSubtotal() {
        return subtotal;
    }
    public double getIgv() {
        return igv;
    }
    public double getTotal() {
        return total;
    }
    public void setIdComprobante(int idComprobante) {
        this.idComprobante = idComprobante;
    }
    public void setTipoDocumento(TipoComprobante tipoDocumento) {
        this.tipoDocumento = tipoDocumento;
    }
    public void setRuc(String ruc) {
        this.ruc=ruc;
    }
    public void setRazonSocial(String razonSocial) {
        this.razonSocial = razonSocial;
    }
    public void setNumeroSerial(String numeroSerial) {
        this.numeroSerial = numeroSerial;
    }
    public void setFechaEmision(Date fechaEmision) {
        this.fechaEmision = fechaEmision;
    }
    public void setMontoTotal(double montoTotal) {
        this.montoTotal = montoTotal;
    }
    public void setSubtotal(double subtotal) {
        this.subtotal = subtotal;
    }
    public void setIgv(double igv) {
        this.igv = igv;
    }
    public void setTotal(double total) {
        this.total = total;
    }
}

File: web/EconomixSolution/EconomixModel/Model/operaciones/ComprobantePago.cs
using System.Text.Json.Serialization;

namespace EconomixModel.Model;

public class ComprobantePago
{
    [JsonPropertyName("idComprobante")]
    public int IdComprobante { get; set; }

    [JsonPropertyName("tipoDocumento")]
    public string TipoDocumento { get; set; } = "FACTURA";

    [JsonPropertyName("RUCProveedor")]
    public string? RucProveedor { get; set; }

    [JsonPropertyName("razonSocial")]
    public string? RazonSocial { get; set; }

    [JsonPropertyName("numeroSerial")]
    public string? NumeroSerial { get; set; }

    [JsonPropertyName("fechaEmision")]
    public DateTime? FechaEmision { get; set; }

    [JsonPropertyName("montoTotal")]
    public double MontoTotal { get; set; }

    [JsonPropertyName("tipoCambio")]
    public double TipoCambio { get; set; } = 1.0;

    [JsonPropertyName("montoConvertido")]
    public double MontoConvertido { get; set; }

    [JsonPropertyName("nombreArchivoComprobante")]
    public string NombreArchivoComprobante { get; set; } = string.Empty;

    [JsonPropertyName("subtotal")]
    public double Subtotal { get; set; }

    [JsonPropertyName("igv")]
    public double Igv { get; set; }

    [JsonPropertyName("total")]
    public double Total { get; set; }

    [JsonPropertyName("estado")]
    public string Estado { get; set; } = "POR_REVISAR";

    [JsonPropertyName("solicitud")]
    public SolicitudGasto? Solicitud { get; set; }

    [JsonPropertyName("moneda")]
    public Moneda? Moneda { get; set; }
}
